include/oneapi/tbb/detail/_config.h

```diff
@@ -206,6 +210,11 @@
     #define __TBB_USE_OPTIONAL_RTTI (__GXX_RTTI || __RTTI || __INTEL_RTTI__)
 #endif
 
+/** Address sanitizer detection **/
+#if __SANITIZE_ADDRESS__ || (__clang__ && __has_feature(address_sanitizer))
+#define __TBB_SANITIZE_ADDRESS 1
+#endif
+
```

Comment:
Please move the definition from tests config : https://github.com/oneapi-src/oneTBB/blob/3a6580652d2b697823f38669fb4685da812df13c/test/common/config.h#L88-L94